### PR TITLE
[WNMGDS-2838] Update all themes to make inline tool tips black

### PR DESCRIPTION
**Pull request by jack-ryan-nava-pbc**
## Summary

[Ticket](https://jira.cms.gov/browse/WNMGDS-2838)
Changed the value pointed to in the tokens for '--tooltip-trigger__color` class. 

## How to test

1. Pull down the branch locally and build
2. Confirm that for each theme the color of the tooltip underline and text matches the color of the surrounding text. 

## Verify Screenshots look ok

cms_tooltip
cms_tooltip_hover
core_tooltip
core_tooltip_hover
healthcare_tooltip
healthcare_tooltip_hover
medicare_tooltip
medicare_tooltip_hover

## Checklist

- [X] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### If this is a change to design:

- [x] If visual regression image references have been changed, design **MUST** be assigned to review. In this instance, **designer approval is a requirement** before the PR can be merged.

### If this is a change to code:

- [x] Created or updated unit tests to cover any new or modified code
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### CODE REVIEW

File: packages/design-system-tokens/src/tokens/Theme.cmsgov.json [hunk 1]
```diff
@@ -4507,7 +4507,7 @@
       },
       "trigger": {
         "color": {
-          "$value": "{theme.color.primary}",
+          "$value": "{theme.color.black}",
```
Comment (pwolfert) [inline packages/design-system-tokens/src/tokens/Theme.cmsgov.json:4510]: To confirm, did you change these directly in JSON or make the change in Figma and then sync to code?

Comment (jack-ryan-nava-pbc) [reply packages/design-system-tokens/src/tokens/Theme.cmsgov.json:4510]: Changed directly in JSON. I didn't sync anything yet.

**Review by zarahzachz — COMMENTED** [review]
I don't know how possible this is with Figma tokens, but should this value actually be `currentColor` so the tooltip text/underline matches its surrounding text color? 

I don't know if the goal is to make the tooltip match its surrounding text or to explicitly make it black for all scenarios (including the inverse background - if that's used at all, I don't know)

**Comment by jack-ryan-nava-pbc** [comment]
> I don't know how possible this is with Figma tokens, but should this value actually be `currentColor` so the tooltip text/underline matches its surrounding text color?
> 
> I don't know if the goal is to make the tooltip match its surrounding text or to explicitly make it black for all scenarios (including the inverse background - if that's used at all, I don't know)

@zarahzachz I think this is a good callout. Looks like all of the Design Systems are using `color--base` in the text inherited from the `<body>`. I should be able to reference that variable in the themes via `theme.color.base` and give that a shot.

File: tests/browser/snapshots/stories/components-icons--hhs-logo-medicare--firefox.png
Comment (zarahzachz) [inline tests/browser/snapshots/stories/components-icons--hhs-logo-medicare--firefox.png:1]: Was this icon supposed to change?

File: tests/browser/snapshots/stories/reset-form-elements--form-elements-medicare--webkit.png
Comment (zarahzachz) [inline tests/browser/snapshots/stories/reset-form-elements--form-elements-medicare--webkit.png:1]: Looks like a sizing change on this one too

File: tests/browser/snapshots/stories/components-icons--hhs-logo-medicare--firefox.png
Comment (jack-ryan-nava-pbc) [reply tests/browser/snapshots/stories/components-icons--hhs-logo-medicare--firefox.png:1]: Nope. That's unexpected. Hmm...

File: tests/browser/snapshots/stories/reset-form-elements--form-elements-medicare--webkit.png
Comment (jack-ryan-nava-pbc) [reply tests/browser/snapshots/stories/reset-form-elements--form-elements-medicare--webkit.png:1]: also wut

**Comment by zarahzachz** [comment]
> > I don't know how possible this is with Figma tokens, but should this value actually be `currentColor` so the tooltip text/underline matches its surrounding text color?
> > I don't know if the goal is to make the tooltip match its surrounding text or to explicitly make it black for all scenarios (including the inverse background - if that's used at all, I don't know)
> 
> @zarahzachz I think this is a good callout. Looks like all of the Design Systems are using `color--base` in the text inherited from the `<body>`. I should be able to reference that variable in the themes via `theme.color.base` and give that a shot.

Oh, my recommendation wasn't to change it to `color--base` - it was to change it to `currentColor`, which I don't know is possible with the current token setup (Figma only supports a subset of CSS properties). 

`currentColor` takes the color of whatever its parent has defined, meaning if this text were changed using a utility class like `ds-u-color--error`, the entire string (including the tooltip) would render red. I've attached a video explaining what I mean: 

https://github.com/user-attachments/assets/231545e0-b7cd-4fef-9bd8-baf13fc82689

I don't think Figma supports `currentColor`, which may mean removing this token from the project and hardcoding it to `currentColor` (we do this on other components like SvgIcon fill). Since this token is only used in this one instance, I think an argument could be made in favor of deprecating the token. That would make this a breaking change, but since this is meant to go out in v12 I don't think that's an issue. Just things to talk about with your team.

**Comment by jack-ryan-nava-pbc** [comment]
Noting from discussion as of today:
1. We are continuing with these changes
2. We will investigate removing the tokens associated with Tooltips as part of our future Tooltip discovery work. That work will allow us to use `currentColor`

File: packages/ds-medicare-gov/src/styles/components/_Tooltip.scss [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
-  text-decoration-color: var(--color-primary);
+  text-decoration-color: var(--tooltip-trigger__color);
   text-decoration-style: dashed;
   text-decoration-thickness: var(--link__text-decoration-thickness);
   text-underline-offset: var(--link__text-decoration-offset);
 
   &:hover {
     color: var(--color-black);
-    text-decoration-color: var(--color-primary-darker);
+    text-decoration-color: var(--color-black);
```
Comment (pwolfert) [inline packages/ds-medicare-gov/src/styles/components/_Tooltip.scss:16]: Is this a temporary decision? Should it be flagged to revisit later?

Comment (pwolfert) [reply packages/ds-medicare-gov/src/styles/components/_Tooltip.scss:16]: Also, I found through testing that you've overridden the hover color for medicare but not for any of the other themes. Is that intentional?

Comment (jack-ryan-nava-pbc) [reply packages/ds-medicare-gov/src/styles/components/_Tooltip.scss:16]: No that was a whoopsie. Thank you for finding that. 

I need to document in the ticket the intent behind these changes and that they will likely be reviewed and possibly changed as we embark on the tooltip discovery work. I think at this moment I don't know if these changes are temporary in such a way that I know for a fact they will be removed and when. So given that, I think documentation in the ticket should be sufficient, but also happy to add a comment to the SCSS file.

Comment (pwolfert) [reply packages/ds-medicare-gov/src/styles/components/_Tooltip.scss:16]: 👍 It's up to you how you want to document it. I just wanted you to be able to remember it.

**Review by malloryiden — APPROVED** [review]
🎉

File: packages/design-system-tokens/src/tokens/Theme.cmsgov.json [hunk 3]
```diff
@@ -4507,7 +4507,7 @@
       },
       "trigger": {
         "color": {
-          "$value": "{theme.color.primary}",
```
Comment (kim-cmsds) [inline packages/design-system-tokens/src/tokens/Theme.cmsgov.json:4510]: @jack-ryan-nava-pbc have you run the sync to figma script yet?

**Review by kim-cmsds — APPROVED** [review]
Sounds like merging first, then running `syncTokensToFigma` is the path forward for making sure these tokens are updated in Figma. Thanks @jack-ryan-nava-pbc!

**Pull request merged**